jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryClientTest.java

```diff
@@ -211,6 +213,82 @@ public void testConfigureBasicAuth()
+            + "        \"digest\": \"sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7\"\n"
+            + "    }\n"
+            + "}";
+
+    registry = new TestWebServer(false, Arrays.asList(manifestResponse), 1);
+    RegistryClient registryClient = createRegistryClient(null);
+    ManifestAndDigest<?> manifestAndDigest = registryClient.pullManifest("image-tag");
+
+    Assert.assertEquals(
+        "sha256:6b61466eabab6e5ffb68ae2bd9b85c789225540c2ac54ea1f71eb327588e8946",
+        manifestAndDigest.getDigest().toString());
+    Assert.assertTrue(manifestAndDigest.getManifest() instanceof V22ManifestTemplate);
+    V22ManifestTemplate manifest = (V22ManifestTemplate) manifestAndDigest.getManifest();
+    Assert.assertEquals(2, manifest.getSchemaVersion());
+    Assert.assertEquals(
+        "application/vnd.docker.distribution.manifest.v2+json", manifest.getManifestMediaType());
+    Assert.assertEquals(
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+        manifest.getContainerConfiguration().getDigest().toString());
+    Assert.assertEquals(7023, manifest.getContainerConfiguration().getSize());
+    Assert.assertThat(
+        registry.getInputRead(),
+        CoreMatchers.containsString("GET /v2/foo/bar/manifests/image-tag "));
+  }
+
+  @Test
+  public void testPullManifest_manifestList()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException,
+          RegistryException {
+    String manifestResponse =
+        "HTTP/1.1 200 OK\nContent-Length: 403\n\n{\n"
+            + "  \"schemaVersion\": 2,\n"
+            + "  \"mediaType\": \"application/vnd.docker.distribution.manifest.list.v2+json\",\n"
+            + "  \"manifests\": [\n"
+            + "    {\n"
+            + "      \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "      \"size\": 7143,\n"
+            + "      \"digest\": \"sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f\",\n"
+            + "      \"platform\": {\n"
+            + "        \"architecture\": \"amd64\",\n"
+            + "        \"os\": \"linux\"\n"
+            + "      }\n"
+            + "    }\n"
+            + "  ]\n"
+            + "}";
+
+    registry = new TestWebServer(false, Arrays.asList(manifestResponse), 1);
+    RegistryClient registryClient = createRegistryClient(null);
+    ManifestAndDigest<?> manifestAndDigest = registryClient.pullManifest("image-tag");
+
+    Assert.assertEquals(
+        "sha256:a340fa38667f765f8cfd79d4bc684ec8a6f48cdd63abfe36e109f4125ee38488",
+        manifestAndDigest.getDigest().toString());
+    Assert.assertTrue(manifestAndDigest.getManifest() instanceof V22ManifestListTemplate);
+    V22ManifestListTemplate manifestList =
+        (V22ManifestListTemplate) manifestAndDigest.getManifest();
+    Assert.assertEquals(2, manifestList.getSchemaVersion());
+    Assert.assertEquals(
```

Comment:
`getDigestsForPlatform()` returns a `List` object. For complex objects that have multiple other objects/fields/values, often their `toString()` is more or less an internal/debug string representation of the objects. That is, theoretically the string form generated by `List.toString()` is not canonical and may change in future Java versions. For verifying the elements in a list, you would rather do

```java
assertEquals(
    Arrays.asList("sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f"), 
    manifestList.getDigestsForPlatform("amd64", "linux"));
```

Reply:
done